Code/Tools/ProjectManager/Source/CreateAGemScreen.cpp

```diff
@@ -0,0 +1,433 @@
+            setTabPosition(QTabWidget::West);
+            setTabShape(QTabWidget::TabShape::Rounded);
+        }
+    };
+
+    CreateGem::CreateGem(QWidget* parent)
+        : ScreenWidget(parent)
+    {
+        QVBoxLayout* vLayout = new QVBoxLayout(this);
+        vLayout->setSpacing(0);
+        vLayout->setContentsMargins(0, 0, 0, 0);
+
+        m_tabWidget = new TabWidget(this);
+        m_tabWidget->setContentsMargins(0, 0, 0, 0);
+        
+        m_tabWidget->setObjectName("createGemTab");
+        m_tabWidget->addTab(CreateGemSetupScrollArea(), tr("1. Gem Setup"));
+        m_tabWidget->addTab(CreateGemDetailsScrollArea(), tr("2. Gem Details"));
+        m_tabWidget->addTab(CreateGemCreatorScrollArea(), tr("3. Creator Details"));
+        m_tabWidget->tabBar()->setEnabled(false);
+        vLayout->addWidget(m_tabWidget);
+
+        QFrame* footerFrame = new QFrame(this);
+        footerFrame->setObjectName("createAGemFooter");
+        m_backNextButtons = new QDialogButtonBox(this);
+        m_backNextButtons->setObjectName("footer");
+        QVBoxLayout* footerLayout = new QVBoxLayout();
+        footerLayout->setContentsMargins(0, 0, 0, 0);
+        footerFrame->setLayout(footerLayout);
+        footerLayout->addWidget(m_backNextButtons);
+        vLayout->addWidget(footerFrame);
+
+        m_backButton = m_backNextButtons->addButton(tr("Back"), QDialogButtonBox::RejectRole);
+        m_backButton->setProperty("secondary", true);
+        m_nextButton = m_backNextButtons->addButton(tr("Next"), QDialogButtonBox::ApplyRole);
+
+        connect(m_tabWidget, &TabWidget::currentChanged, this, &CreateGem::UpdateNextButtonToCreate);
+        connect(m_backButton, &QPushButton::clicked, this, &CreateGem::HandleBackButton);
+        connect(m_nextButton, &QPushButton::clicked, this, &CreateGem::HandleNextButton);
+
+        setObjectName("createAGemBody");
+        setLayout(vLayout);
+    }
+
+    void CreateGem::LoadButtonsFromGemTemplatePaths(QVBoxLayout* gemSetupLayout)
+    {
+        m_radioButtonGroup = new QButtonGroup();
+
+        auto templatesResult = PythonBindingsInterface::Get()->GetGemTemplates();
+        if (templatesResult.IsSuccess() && !templatesResult.GetValue().isEmpty())
+        {
+            m_gemTemplates = templatesResult.GetValue();
+            for (int index = 0; index < m_gemTemplates.size(); ++index)
+            {
+                ProjectTemplateInfo gemTemplate = m_gemTemplates.at(index);
+
+                QRadioButton* button = new QRadioButton(gemTemplate.m_displayName);
+                button->setObjectName("createAGem");
+                m_radioButtonGroup->addButton(button, index);
+
+                QLabel* buttonSubtext = new QLabel();
+                buttonSubtext->setObjectName("createAGemRadioButtonSubtext");
+                buttonSubtext->setText(gemTemplate.m_summary);
+                if (index == 0)
+                {
+                    button->click();
+                }
```

Comment:
so I guess this is necessary?

Reply:
It’s necessary to manually make sure at least one of the buttons is selected even though the group is exclusive. I left this as it was but there could be a better way of doing this.